src/Domain/Creators/TimeDependence/ScalingAndZRotation.hpp

```diff
@@ -0,0 +1,198 @@
+ * y &\to& x \sin \alpha(t) + y \cos \alpha(t)\text{,}
+ * \f}
+ * where \f$\alpha(t)\f$ is a `domain::FunctionsOfTime::FunctionOfTime`. For 3
+ * spatial dimensions, \f$z \to z\f$, and the rotation is implemented as a
+ * product of the 2D rotation and an identity map. The rotation is undefined
+ * (and therefore unimplemented here) for 1 spatial dimension.
+ *
+ * The expansion is done by the
+ * `domain::CoordinateMaps::TimeDependent::CubicScale` map. A linear
+ * radial scaling can be used by specifying the `UseLinearScaling` bool.
+ */
+template <size_t MeshDim>
+class ScalingAndZRotation final : public TimeDependence<MeshDim> {
+  static_assert(
+      MeshDim > 1,
+      "ScalingAndZRotation<MeshDim> undefined for MeshDim == 1");
+
+ private:
+  using Identity = domain::CoordinateMaps::Identity<1>;
+  using Rotation = domain::CoordinateMaps::TimeDependent::Rotation<2>;
+  using CubicScaleMap =
+      domain::CoordinateMaps::TimeDependent::CubicScale<MeshDim>;
+
+ public:
+  using maps_list = tmpl::list<
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, Rotation>,
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, CubicScaleMap>,
+      domain::CoordinateMap<
+          Frame::Grid, Frame::Inertial,
+          CoordinateMaps::TimeDependent::ProductOf2Maps<Rotation, Identity>>>;
+
+  static constexpr size_t mesh_dim = MeshDim;
+
+  /// \brief The initial time of the function of time.
+  struct InitialTime {
+    using type = double;
+    static constexpr Options::String help = {
+        "The initial time of the function of time"};
+  };
+  /// \brief The \f$x\f$-, \f$y\f$-, and \f$z\f$-velocity.
+  struct AngularVelocity {
+    using type = double;
+    static constexpr Options::String help = {
+        "The angular velocity of the map."};
+  };
+
+  /// \brief The outer boundary or pivot point of the
+  /// `domain::CoordinateMaps::TimeDependent::CubicScale` map
+  struct OuterBoundary {
+    using type = double;
+    static constexpr Options::String help = {
+        "Outer boundary or pivot point of the map"};
+  };
+  /// \brief The initial values of the expansion factors.
+  struct InitialExpansion {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {
+        "Expansion values at initial time."};
+  };
+  /// \brief The velocity of the expansion factors.
+  struct Velocity {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {"The rate of expansion."};
+  };
+  /// \brief The acceleration of the expansion factors.
+  struct Acceleration {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {"The acceleration of expansion."};
+  };
+  /// \brief Whether to use linear scaling or cubic scaling.
+  struct UseLinearScaling {
+    using type = bool;
+    static constexpr Options::String help = {
+        "Whether or not to turn on cubic scaling."};
+  };
+
+  using MapForComposition = detail::generate_coordinate_map_t<
```

Comment:
The name for this alias is ok here because this is actually a composition, but in all the other `TimeDependence`s it should be changed because there is only one map given to `generate_coordinate_map_t`.